table/block_based_table_reader.cc

```diff
@@ -702,30 +704,27 @@ Status BlockBasedTable::Open(const ImmutableCFOptions& ioptions,
   }
 
   // Read the range del meta block
-  // TODO(wanning&andrewkr): cache range delete tombstone block
   bool found_range_del_block;
-  BlockHandle range_del_handle;
   s = SeekToRangeDelBlock(meta_iter.get(), &found_range_del_block,
-                          &range_del_handle);
+                          &rep->range_del_handle);
   if (!s.ok()) {
     Log(InfoLogLevel::WARN_LEVEL, rep->ioptions.info_log,
         "Error when seeking to range delete tombstones block from file: %s",
         s.ToString().c_str());
```

Comment:
wow this is so confusing, I was wondering for a while how the BlockHandle fields became nonzero, thus making range_del_handle non-null, when the meta-block doesn't exist in the file. my idea was they should be zero since we use the default constructor, the member variables default to zero (https://github.com/facebook/rocksdb/blob/17f76fc5647ce1a06055c3eae8e5c8055749566f/table/format.h#L68-L69), and the call to SeekToRangeDelBlock() appears to not overwrite those members when the block is not found.

it turned out we implement the default constructor to use nonzero values (https://github.com/facebook/rocksdb/blob/17f76fc5647ce1a06055c3eae8e5c8055749566f/table/format.h#L245-L248) so actually the default values are meaningless.

so I think the first thing we should do is initialize range_del_block to BlockHandle::NullBlockHandle(). for the case you're asking about, yes we should reset range_del_handle, I will update it.